### PR TITLE
refactor(devtools-view): Refactor tree helper structure to reduce nested indentation

### PR DESCRIPTION
**Pull request by Josmithr**
FluentUI's experimental Tree component is very strange. It requires setting up an outer `Tree` component context, which is really (semantically) a "Tree Context". Inside of this context, an actual tree is created using the `TreeItem` component (very unintuitive), and the inline contents of that tree (the "header") are specified via the `TreeItemLayout` component.

Because this is so counterintuitive, our current implementation introduces a lot of ultimately unneeded component hierarchy, which introduces a lot of unwanted and unpredictable indentation. 

These changes are aimed at cleaning up that indentation, which it accomplishes by re-structuring our app tree. Since the setup is unintuitive, I went ahead and made our tree helpers an implementation detail of our data-visualization components directory. If we want to try and generalize this later, we can, but for now I am opting out of sharing.

This PR also:
* Encapsulates `node` prop accepted by the various tree visualization components into a shared interface.
* Adds `@fluentui/react-components/unstable` to an allow-list in the eslint config so we don't have to disable line-by-line.
* Removes usages of `nodeKind` in user-facing visualization (this data is an implementation detail of the visualization tree format).
* Adds labels to unknown data type tree nodes

image

### CODE REVIEW

**Comment by jikim-msft** [comment]
In addition, both `Private Container` and `Shared Container` won't load, if I add `aria-label` to the `<FluentTree>` tag from `DataObjectsView.tsx`.

**Comment by Josmithr** [comment]
> image
> 
> I think the representation of `unknown-data` and `unknown-shared-object` could be misleading when compared to the data (such as `shared-text` and `shared-counter`) which are stored in the same level within the `rootMap`.
> 
> image
> 
> When we first populate the `rootMap`, every key except the `emoji-matrix` and `record-value` are stored in same level, thus have a format of `label (nodeTypeMetadata?): value` pair. However, we represent `unknown` objects in the format of:
> 
> * `>` `label (nodeTypeMetadata?)`
>   
>   * `value`
> 
> Since `unknown` nodes are not nested key-value pair inside the `rootMap` (for example `record-value`), I personally think it makes more sense to format like:
> 
> * `unknown-data: Unrecognized kind of data.`
> * `unknown-shared-object (mock-shared-object-type): Unrecognized kind of Fluid Object.`
> 
> I might be missing something, so please feel free to share your thoughts 😃.

That's fair. I think you're right. I'll update them to put their content inline.

**Comment by Josmithr** [comment]
> > > In addition, both `Private Container` and `Shared Container` won't load, if I add `aria-label` to the `<FluentTree>` tag from `DataObjectsView.tsx`.
> > 
> > 
> > @jikim-msft can you elaborate? I'm not sure what you mean.
> 
> image
> 
> https://react.fluentui.dev/?path=/docs/preview-components-tree-tree--default
> 
> As suggested in the official documentation, `Tree` must include either `aria-label` or `aria-labelledby` as its property for accessibility. I tried adding descriptions like ` to abide by the recommendation.
> 
> image
> 
> However, adding `aria-label` would crash the rendering of the container. Interested to know the reason.

Interesting. I was able to repro this as well. I suspect it's an ID uniqueness issue. I included the containerId value in the aria string to attempt to differentiate the two containers' data views, and that allowed one of them to render correctly, but not the other.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/CommonInterfaces.ts [hunk 1]
```diff
@@ -3,11 +3,30 @@
...
+	 * This will commonly be the property name, map key, etc. under which the data was stored in the
+	 * corresponding application's data model.
+	 */
 	label: string;
 }
+
+/**
+ * Base props interface for components displaying {@link @fluid-experimental/devtools-core#VisualTreeNode}s.
+ */
+export interface DataVisalizationTreeProps<TNode extends VisualNodeBase = VisualNode>
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/components/data-visualization/CommonInterfaces.ts:26]:
```suggestion
export interface DataVisualizationTreeProps<TNode extends VisualNodeBase = VisualNode>
```

**Review by alexvy86 — COMMENTED** [review]
I mostly follow the changes and don't see anything obviously wrong, but I think @jikim-msft has a better grasp on this so I'll leave the approval to him.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/CommonInterfaces.ts (hunk 1 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/CommonInterfaces.ts:26]: 🤦‍♂️

**Comment by Josmithr** [comment]
> > > > In addition, both `Private Container` and `Shared Container` won't load, if I add `aria-label` to the `<FluentTree>` tag from `DataObjectsView.tsx`.
> > > 
> > > 
> > > @jikim-msft can you elaborate? I'm not sure what you mean.
> > 
> > 
> > image
> > https://react.fluentui.dev/?path=/docs/preview-components-tree-tree--default
> > As suggested in the official documentation, `Tree` must include either `aria-label` or `aria-labelledby` as its property for accessibility. I tried adding descriptions like ` to abide by the recommendation.
> > image
> > However, adding `aria-label` would crash the rendering of the container. Interested to know the reason.
> 
> Interesting. I was able to repro this as well. I suspect it's an ID uniqueness issue. I included the containerId value in the aria string to attempt to differentiate the two containers' data views, and that allowed one of them to render correctly, but not the other.

@jikim-msft After merging in most recent main, I'm no longer seeing this issue. Are you still able to repro?

**Comment by rohandubal** [comment]
Visually looks good on my end; like Alejandro I will leave some structural/best approval for Ji.

**Review by rohandubal — APPROVED** [review]

**Comment by jikim-msft** [comment]
> > > > > In addition, both `Private Container` and `Shared Container` won't load, if I add `aria-label` to the `<FluentTree>` tag from `DataObjectsView.tsx`.
> > > > 
> > > > 
> > > > @jikim-msft can you elaborate? I'm not sure what you mean.
> > > 
> > > 
> > > image
> > > https://react.fluentui.dev/?path=/docs/preview-components-tree-tree--default
> > > As suggested in the official documentation, `Tree` must include either `aria-label` or `aria-labelledby` as its property for accessibility. I tried adding descriptions like `to abide by the recommendation. image However, adding`aria-label` would crash the rendering of the container. Interested to know the reason.
> > 
> > 
> > Interesting. I was able to repro this as well. I suspect it's an ID uniqueness issue. I included the containerId value in the aria string to attempt to differentiate the two containers' data views, and that allowed one of them to render correctly, but not the other.
> 
> @jikim-msft After merging in most recent main, I'm no longer seeing this issue. Are you still able to repro?

Everything's working fine now! 😄 Thanks for the work Josh!

**Review by jikim-msft — APPROVED** [review]

**Pull request merged**